### PR TITLE
Traducido referecne/compound_stmts {compound_stmts.po}

### PR DESCRIPTION
**Pull request by mayuti**
Closes #496

### CODE REVIEW

**Comment by mayuti** [comment]
Closes #<496>

File: reference/compound_stmts.po [hunk 1]
```diff
@@ -76,20 +105,28 @@ msgid ""
...
 "The formatting of the grammar rules in the following sections places each "
 "clause on a separate line for clarity."
 msgstr ""
+"El formato de las reglas gramaticales en las siguientes secciones coloca "
+"cada cláusula en una línea separada para mayor claridad."
 
 #: ../Doc/reference/compound_stmts.rst:82
 msgid "The :keyword:`!if` statement"
-msgstr ""
+msgstr "La sentencia :keyword:`!If`"
```
Comment (cmaureir) [inline reference/compound_stmts.po:125]:
```suggestion
msgstr "La sentencia :keyword:`!if`"
```

File: reference/compound_stmts.po [hunk 2]
```diff
@@ -158,13 +223,21 @@ msgid ""
...
 
 #: ../Doc/reference/compound_stmts.rst:177
 msgid ""
 "The for-loop makes assignments to the variables in the target list. This "
 "overwrites all previous assignments to those variables including those made "
 "in the suite of the for-loop::"
 msgstr ""
+"El ciclo for realiza asignaciones a las variables en la lista. Esto "
+"sobrescribe todas las asignaciones anteriores a esas variables, incluidas "
+"las realizadas en la suite del ciclo for::"
```
Comment (cmaureir) [inline reference/compound_stmts.po:240]:
```suggestion
"las realizadas en la suite del bucle for::"
```

File: reference/compound_stmts.po [hunk 3]
```diff
@@ -174,6 +247,11 @@ msgid ""
 "integers suitable to emulate the effect of Pascal's ``for i := a to b do``; "
 "e.g., ``list(range(3))`` returns the list ``[0, 1, 2]``."
 msgstr ""
+"Los nombres en la lista no se eliminan cuando finaliza el ciclo, pero si la "
+"secuencia está vacía, el ciclo no les habrá asignado nada. Sugerencia: la "
+"función incorporada :func:`range` devuelve un iterador de enteros adecuado "
+"para emular el efecto de Pascal``for i := a to b do``; por ejemplo, "
```
Comment (cmaureir) [inline reference/compound_stmts.po:253]:
```suggestion
"para emular el efecto de Pascal ``for i := a to b do``; por ejemplo, "
```

File: reference/compound_stmts.po [hunk 4]
```diff
@@ -174,6 +247,11 @@ msgid ""
...
+"``list(range (3))`` devuelve la lista ``[0, 1, 2]``."
```
Comment (cmaureir) [inline reference/compound_stmts.po:254]:
```suggestion
"``list(range(3))`` devuelve la lista ``[0, 1, 2]``."
```

File: reference/compound_stmts.po [hunk 5]
```diff
@@ -189,16 +267,30 @@ msgid ""
...
 msgid "The :keyword:`!try` statement"
-msgstr ""
+msgstr "La sentencia :keyword:`!try`"
 
 #: ../Doc/reference/compound_stmts.rst:234
 msgid ""
 "The :keyword:`try` statement specifies exception handlers and/or cleanup "
 "code for a group of statements:"
 msgstr ""
+"La sentencia :keyword:`try` es especifica para gestionar excepciones y/o "
```
Comment (cmaureir) [inline reference/compound_stmts.po:292]:
```suggestion
"La sentencia :keyword:`try` es especifica para gestionar excepciones o "
```

File: reference/compound_stmts.po [hunk 6]
```diff
@@ -214,12 +306,27 @@ msgid ""
 "exception if it is the class or a base class of the exception object or a "
 "tuple containing an item compatible with the exception."
 msgstr ""
+"La/s cláusula/s :keyword:`except` especifican uno o más gestores de "
+"excepciones. Cuando no se produce ninguna excepción en la cláusula :keyword:"
+"`try`, no se ejecuta ningún gestor de excepción. Cuando se produce una "
+"excepción en la suite :keyword: `!try`, se inicia la búsqueda de un gestor "
```
Comment (cmaureir) [inline reference/compound_stmts.po:312]:
```suggestion
"excepción en la suite :keyword:`!try`, se inicia la búsqueda de un gestor "
```

File: reference/compound_stmts.po [hunk 7]
```diff
@@ -308,6 +461,10 @@ msgid ""
 "finally` statement, the :keyword:`finally` clause is also executed 'on the "
 "way out.'"
 msgstr ""
+"Cuando se ejecuta una sentencia :keyword:`return`, :keyword:`break` o :"
+"keyword:`continue` en la suite :keyword:`try` de un :keyword: `!try` ... \\ "
```
Comment (cmaureir) [inline reference/compound_stmts.po:465]:
```suggestion
"keyword:`continue` en la suite :keyword:`try` de un :keyword:`!try` ...\\ "
```

File: reference/compound_stmts.po [hunk 8]
```diff
@@ -308,6 +461,10 @@ msgid ""
...
+"La sentencia :keyword:`!finally`, la cláusula :keyword:`finally` también se "
```
Comment (cmaureir) [inline reference/compound_stmts.po:466]:
```suggestion
"la sentencia :keyword:`!finally`, la cláusula :keyword:`finally` también se "
```

File: reference/compound_stmts.po [hunk 9]
```diff
@@ -341,36 +508,50 @@ msgid ""
 "This allows common :keyword:`try`...\\ :keyword:`except`...\\ :keyword:"
 "`finally` usage patterns to be encapsulated for convenient reuse."
 msgstr ""
+"La sentencia: keyword:`with` se usa para ajustar la ejecución de un bloque "
+"con métodos definidos por un administrador de contexto (ver sección :ref:"
+"`context-managers`). Esto permite que los patrones de uso comunes :keyword:"
+"`try` ... \\ :keyword:`except` ... \\ :keyword:`finally` se encapsulen para "
```
Comment (cmaureir) [inline reference/compound_stmts.po:514]:
```suggestion
"`try` ...\\ :keyword:`except` ...\\ :keyword:`finally` se encapsulen para "
```

File: reference/compound_stmts.po [hunk 10]
```diff
@@ -341,36 +508,50 @@ msgid ""
...
+"una reutilización conveniente."
 
 #: ../Doc/reference/compound_stmts.rst:397
 msgid ""
 "The execution of the :keyword:`with` statement with one \"item\" proceeds as "
 "follows:"
 msgstr ""
+"La ejecución de la sentencia: keyword:`with` con un \"item\" se realiza de "
```
Comment (cmaureir) [inline reference/compound_stmts.po:522]:
```suggestion
"La ejecución de la sentencia :keyword:`with` con un \"item\" se realiza de "
```

File: reference/compound_stmts.po [hunk 11]
```diff
@@ -458,12 +661,20 @@ msgid ""
...
+"actual como el espacio de nombres global que se utilizará cuando se llama a "
+"la función."
 
 #: ../Doc/reference/compound_stmts.rst:529
 msgid ""
 "The function definition does not execute the function body; this gets "
 "executed only when the function is called. [#]_"
 msgstr ""
+"La definición de la función no ejecuta el cuerpo de la función; esto se "
+"ejecuta solo cuando se llama a la función. [#] _"
```
Comment (cmaureir) [inline reference/compound_stmts.po:677]:
```suggestion
"ejecuta solo cuando se llama a la función. [#]_"
```

File: reference/compound_stmts.po [hunk 12]
```diff
@@ -475,17 +686,26 @@ msgid ""
...
 msgid "is roughly equivalent to ::"
-msgstr ""
+msgstr "es más o menos equivalente a ::"
 
 #: ../Doc/reference/compound_stmts.rst:551
 msgid ""
 "except that the original function is not temporarily bound to the name "
 "``func``."
 msgstr ""
+"excepto que la función original no está vinculada temporalmente al nombre `` "
```
Comment (cmaureir) [inline reference/compound_stmts.po:707]:
```suggestion
"excepto que la función original no está vinculada temporalmente al nombre ``"
```

File: reference/compound_stmts.po [hunk 13]
```diff
@@ -497,6 +717,14 @@ msgid ""
 "up until the \"``*``\" must also have a default value --- this is a "
 "syntactic restriction that is not expressed by the grammar."
 msgstr ""
+"Cuando uno o más :term:`parameters <parameter>` tienen la forma *parameter* "
+"``=`` *expression*, se dice que la función tiene \"valores de parámetros "
+"predeterminados\". Para un parámetro con un valor predeterminado, el "
+"correspondiente :term:`argument` puede omitirse desde una llamada, en cuyo "
+"caso se sustituye el valor predeterminado del parámetro. Si un parámetro "
+"tiene un valor predeterminado, todos los parámetros siguientes hasta el \"` "
+"`*` `\" también deben tener un valor predeterminado --- esta es una "
```
Comment (cmaureir) [inline reference/compound_stmts.po:726]:
```suggestion
"tiene un valor predeterminado, todos los parámetros siguientes hasta el \"`"
"`*``\" también deben tener un valor predeterminado --- esta es una "
```

File: reference/compound_stmts.po [hunk 14]
```diff
@@ -524,6 +763,18 @@ msgid ""
 "Parameters after \"``*``\" or \"``*identifier``\" are keyword-only "
 "parameters and may only be passed used keyword arguments."
 msgstr ""
+"La semántica de llamadas de función se describe con más detalle en la "
+"sección :ref:`calls`. Una llamada a la función siempre asigna valores a "
+"todos los parámetros mencionados en la lista de parámetros, ya sea desde "
+"argumentos de posición, desde argumentos por palabra clave o desde valores "
+"predeterminados. Si está presente la forma \"` `* identifier``\", se "
```
Comment (cmaureir) [inline reference/compound_stmts.po:770]:
```suggestion
"predeterminados. Si está presente la forma \"``*identifier``\", se "
```

File: reference/compound_stmts.po [hunk 15]
```diff
@@ -524,6 +763,18 @@ msgid ""
...
+"inicializa en una tupla que recibe cualquier parámetro posicional en exceso, "
+"por defecto en la tupla vacía. Si el formulario \"` `**identifier``\" está "
```
Comment (cmaureir) [inline reference/compound_stmts.po:772]:
```suggestion
"por defecto en la tupla vacía. Si el formulario \"``**identifier``\" está "
```

File: reference/compound_stmts.po [hunk 16]
```diff
@@ -524,6 +763,18 @@ msgid ""
...
+"presente, se inicializa a una nueva asignación ordenada que recibe cualquier "
+"exceso de argumentos por palabra clave, por defecto a una nueva asignación "
+"vacía del mismo tipo. Los parámetros después de \"` `*` `\" o \"` "
```
Comment (cmaureir) [inline reference/compound_stmts.po:775]:
```suggestion
"vacía del mismo tipo. Los parámetros después de \"``*``\" o \"`"
```

File: reference/compound_stmts.po [hunk 17]
```diff
@@ -541,6 +792,20 @@ msgid ""
 "definition is executed.  In this case annotations may be evaluated in a "
 "different order than they appear in the source code."
 msgstr ""
+"Los parámetros pueden tener :term:`annotation <function annotation>` de la "
+"forma \"``: expression``\" que sigue al nombre del parámetro. Cualquier "
+"parámetro puede tener una anotación, incluso las de la forma ``*identifier`` "
+"o ``** identifier``. Las funciones pueden tener una anotación \"return\" de "
+"la forma \"``->expression``\" después de la lista de parámetros. Estas "
```
Comment (cmaureir) [inline reference/compound_stmts.po:799]:
```suggestion
"la forma \"``-> expression``\" después de la lista de parámetros. Estas "
```

File: reference/compound_stmts.po [hunk 18]
```diff
@@ -631,6 +923,15 @@ msgid ""
 "attribute dictionary.  The class name is bound to this class object in the "
 "original local namespace."
 msgstr ""
+"La suite de la clase se ejecuta en un nuevo marco de ejecución (ver :ref:"
+"`naming`), usando un espacio de nombres local recién creado y el espacio de "
+"nombres global original. (Por lo general, el bloque contiene principalmente "
+"definiciones de funciones). Cuando la suite de la clase finaliza la "
+"ejecución, su marco de ejecución se descarta pero se guarda su espacio de "
+"nombres local. [#] _ Luego se crea un objeto de clase utilizando la lista de "
```
Comment (cmaureir) [inline reference/compound_stmts.po:931]:
```suggestion
"nombres local. [#]_ Luego se crea un objeto de clase utilizando la lista de "
```

File: reference/compound_stmts.po [hunk 19]
```diff
@@ -668,34 +980,48 @@ msgid ""
 "<descriptors>` can be used to create instance variables with different "
 "implementation details."
 msgstr ""
+"** Nota del programador: ** Las variables definidas en la definición de la "
+"clase son atributos de clase; son compartidos por instancias. Los atributos "
+"de instancia se pueden establecer en un método con ``self.name = value``. Se "
+"puede acceder a los atributos de clase e instancia a través de la notación "
+"\"` `self.name``\", y un atributo de instancia oculta un atributo de clase "
```
Comment (cmaureir) [inline reference/compound_stmts.po:987]:
```suggestion
"\"``self.name``\", y un atributo de instancia oculta un atributo de clase "
```

File: reference/compound_stmts.po [hunk 20]
```diff
@@ -705,105 +1031,137 @@ msgid ""
...
+"`async for` y :keyword:`async with` solo se puede usar en los cuerpos de "
+"funciones de corutina."
 
 #: ../Doc/reference/compound_stmts.rst:769
 msgid ""
 "Functions defined with ``async def`` syntax are always coroutine functions, "
 "even if they do not contain ``await`` or ``async`` keywords."
 msgstr ""
+"Las funciones definidas con la sintaxis ``async def`` siempre son funciones "
+"de corutina, incluso si no contienen palabras claves `` await`` o `` async``."
```
Comment (cmaureir) [inline reference/compound_stmts.po:1047]:
```suggestion
"de corutina, incluso si no contienen palabras claves ``await`` o ``async``."
```

File: reference/compound_stmts.po [hunk 21]
```diff
@@ -705,105 +1031,137 @@ msgid ""
...
 
 #: ../Doc/reference/compound_stmts.rst:885
 msgid ""
 "A string literal appearing as the first statement in the class body is "
 "transformed into the namespace's ``__doc__`` item and therefore the class's :"
 "term:`docstring`."
 msgstr ""
+"Una cadena de caracteres literal que aparece como la primera sentencia en el "
+"cuerpo de la clase se transforma en el elemento del espacio de "
+"nombre``__doc__`` y, por lo tanto, de la clase :term:`docstring`."
```
Comment (cmaureir) [inline reference/compound_stmts.po:1167]:
```suggestion
"nombre ``__doc__`` y, por lo tanto, de la clase :term:`docstring`."
```

**Pull request closed without merging**